src/Features/Core/Portable/ChangeSignature/ParameterConfiguration.cs

```diff
@@ -23,42 +32,49 @@ public ParameterConfiguration(IParameterSymbol thisParameter, List<IParameterSym
             SelectedIndex = selectedIndex;
         }
 
-        public static ParameterConfiguration Create(List<IParameterSymbol> parameters, bool isExtensionMethod, int selectedIndex)
+        public static ParameterConfiguration Create(IEnumerable<Parameter?> parameters, bool isExtensionMethod, int selectedIndex)
         {
-            IParameterSymbol thisParameter = null;
-            var parametersWithoutDefaultValues = new List<IParameterSymbol>();
-            var remainingReorderableParameters = new List<IParameterSymbol>();
-            IParameterSymbol paramsParameter = null;
+            var parametersList = parameters.ToList();
+            ExistingParameter? thisParameter = null;
+            var parametersWithoutDefaultValues = ImmutableArray.CreateBuilder<Parameter>();
+            var remainingReorderableParameters = ImmutableArray.CreateBuilder<Parameter>();
+            ExistingParameter? paramsParameter = null;
 
-            if (parameters.Count > 0 && isExtensionMethod)
+            if (parametersList.Count > 0 && isExtensionMethod)
             {
-                thisParameter = parameters[0];
-                parameters.RemoveAt(0);
+                thisParameter = parametersList[0] as ExistingParameter;
+                parametersList.RemoveAt(0);
             }
 
-            if (parameters.Count > 0 && parameters[parameters.Count - 1].IsParams)
+            if (parametersList.Count > 0 && (parametersList[parametersList.Count - 1] as ExistingParameter)?.Symbol.IsParams == true)
             {
-                paramsParameter = parameters[parameters.Count - 1];
-                parameters.RemoveAt(parameters.Count - 1);
+                paramsParameter = parametersList[parametersList.Count - 1] as ExistingParameter;
```

Comment:
why `as` here?  wasn't this certain from the above check?  why not use patterns?

Reply:
Fixed in next review.